docs/internals/contributing/writing-code/coding-style.txt

```diff
@@ -56,6 +48,44 @@ Python style
           """
           pass
 
+Imports
+-------
+
+* Use convenience imports whenever available. For example, do this::
+
+      from django.views.generic import View
+
+  Don't do this::
+
+      from django.views.generic.base import View
+
+* Put imports in five groups: future, standard library, third-party libraries,
+  other Django components, the same component; sort lines in each group
+  alphabetically by the full module name. Place ``import module`` statements
```

Comment:
I've written this fragment choosing a style at random (just looking at at a few files), actually changing it from "before" to "after" in an early commit. Having them sorted out seemed slightly more elegant, but no real arguments on this one.

An imperfect look at the sources:
* a single block with ``from ...`` lines followed by some ``import ...`` lines and then ``from ...`` again: 20 files;
* one or more ``import`` lines followed by two or more ``from`` lines: 48 (some of these are ``import django`` at the beginning of Django imports or are alphabetical by a different coincidence, some others are due to a missing blank line after the standard library block);
* at least two ``from`` lines before an ``import`` line: 17 (again some of these are alphabetical).

Here are the commands I've used, maybe you'll figure out something more accurate:
* ``grep -RPzl '\nfrom[^\n]+\n(import[^\n]+\n)+from' --include=*.py . | wc -l`` (20)
* ``grep -RPzl '\n(import[^\n]+\n)+(from[^\n]+\n){2,}' --include=*.py . | wc -l`` (48)
* ``grep -RPzl '\n(from[^\n]+\n){2,}import' --include=*.py . | wc -l`` (17)

(note that you may drop the ``wc -l`` and replace ``l`` with ``o`` to see the matches in all of these).

I'm not sure if [isort](https://github.com/wrwrwr/django/commit/702be04564d107673c7dee2e116292f612761c03) has a setting for this one -- it seems it puts ``import`` lines before ``from`` lines.